### PR TITLE
Add forklift service

### PR DESCRIPTION
**Pull request by nickgerace**
## Description

This PR adds the forklift service, which "forklifts" data into a data warehouse. Right now, it only "forklifts" one set of data: billing events. That functionality is "opt-in" and the layered load config for the forklift instance must be used to enable data warehouse stream client use.

In addition to these changes, we are publishing fully-formed billing events over NATS Jetstream in four scenarios: the HEAD change set's pointer has changed, the HEAD change set's status has changed, a resource has been created, and a resource has been deleted.

## Additional Changes

- Transform `billing-events-server` into `forklift-server`
- Use and spawn `forklift-server` in anything using the dal test harness
- Move billing events publishing to its own module in the dal
- Move and rename jetstreams streams wrapper to its own module in the dal
- Remove unused and aged dal integration tests
- Create resource metadata and change set status modules with
  corresponding structs and enums in `si-events-rs`
- Adjust relevant `.github` files

### CODE REVIEW

File: lib/dal/src/billing_publish.rs
Comment (britmyerss) [inline lib/dal/src/billing_publish.rs:76]: Are we worried about bloat if we also send the `metadata` itself here?

File: lib/dal/src/billing_publish.rs
Comment (britmyerss) [inline lib/dal/src/billing_publish.rs:161]: great idea to include this!

File: lib/dal/src/billing_publish.rs
Comment (nickgerace) [reply lib/dal/src/billing_publish.rs:76]: Not at all! I think it will be in there if we want it to be. Truthfully, I think I was going to see how big it was first. Worst case we can chunk it.

File: lib/dal/src/billing_publish.rs
Comment (nickgerace) [reply lib/dal/src/billing_publish.rs:161]: I think that was @jkeiser!

File: lib/billing-events-core/src/lib.rs
Comment (jkeiser) [inline lib/billing-events-core/src/lib.rs:3]: Not for this PR, but it looks like we might be able to standardize this (hopefully across the workspace) in Cargo.toml now if we want: [[lints]](https://doc.rust-lang.org/stable/cargo/reference/manifest.html#the-lints-section) was stabilized sometime in the past year.

Comment (nickgerace) [reply lib/billing-events-core/src/lib.rs:3]: Yes! We've been talking about this for awhile. That's the goal! Probably should take some time do it.

The bigger problem isn't the act itself, but ensuring the old parts of the codebase adhere to it.

**Comment by nickgerace** [comment]
This PR is taking on a slightly different form and introducing the new "forklift" service. Still hasn't come out of a drafts yet though.

**Comment by nickgerace** [comment]
Adding @sprutton1 and @johnrwatson to review since this a new service.

**Review by sprutton1 — APPROVED** [review]

**Comment by nickgerace** [comment]
These two minor/cosmetic changes I want to make before merging, as a heads up, but this is still good to go.

File: bin/forklift/Dockerfile
Comment (jkeiser) [inline bin/forklift/Dockerfile:37]: Spelling

Comment (nickgerace) [reply bin/forklift/Dockerfile:37]: Thanks!

Comment (jkeiser) [reply bin/forklift/Dockerfile:37]: (forkflit -> forklift)

File: bin/forklift/src/args.rs
Comment (jkeiser) [inline bin/forklift/src/args.rs:81]: Does the help for these show in the order they appear here? It's typical to have the most program-specific args at the top if so ...

File: lib/dal/src/job/definition/action.rs
Comment (jkeiser) [inline lib/dal/src/job/definition/action.rs:173]: Nice, this should cut down on the number of unnecessary events we send!

File: lib/dal/tests/integration_test/resource_metadata.rs
Comment (jkeiser) [inline lib/dal/tests/integration_test/resource_metadata.rs:13]: Remind me to show you `expected.rs` :)

**Review by jkeiser — APPROVED** [review]
Couple suggestions that don't block merge in the slightest.

File: lib/dal/tests/integration_test/resource_metadata.rs
Comment (nickgerace) [reply lib/dal/tests/integration_test/resource_metadata.rs:13]: I've seen it! It's great work

File: lib/dal/src/job/definition/action.rs
Comment (nickgerace) [reply lib/dal/src/job/definition/action.rs:173]: Yep!

File: bin/forklift/src/args.rs
Comment (nickgerace) [reply bin/forklift/src/args.rs:81]: We usually put them at the bottom of structs

**Pull request merged**
